90 create more coding specific analyzers

priyansh1913 opened the pull request:
<!--
Thank you for contributing! Please fill out this template to help us review your PR.
-->

## Description

Implemented `JavaAnalyzer` and `JavaScriptAnalyzer` classes to extend code analysis capabilities for Java and JavaScript files. Both analyzers count classes, functions/methods, and package imports using regex-based parsing, following the established `CodeFileAnalyzer` pattern.

**Dependencies:** Requires PR #89 which provides `CodeFileAnalyzer` base class

**Closes:** # 90

---

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation added/updated
- [X] Test added/updated
- [ ] Refactoring
- [ ] Performance improvement

---

## Testing

Both analyzers include comprehensive test coverage with dedicated test fixtures.

- [x] `test_JavaAnalyzer` - Validates class count (2), method count (8), and import detection
- [x] `test_JavaScriptAnalyzer` - Validates class count (2), function count (4+), and package imports (react, axios, lodash)
- [x] All 32 tests passing locally

---

## Checklist

- [X] GenAI was used in generating the code and I have performed a self-review of my own code
- [X] I have commented my code where needed
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works and tests are passing locally
- [X] Any dependent changes have been merged and published in downstream modules
- [ ] Any UI changes have been checked to work on desktop, tablet, and/or mobile

---

## Screenshots

> If applicable, add screenshots to help explain your changes

<details>
<summary>Click to expand screenshots</summary>

<!-- Add your screenshots here -->

</details>

sjsikora commented on tests/test_analyzer.py:
```diff
@@ -194,11 +196,11 @@ def test_NaturalLanguageAnalyzer_core_stats(tmp_path: Path):
     statistics about Python files.
     """
-    path = "./tests/resources/example_python.py"
+    path = "./tests/static/example_python.py"
```
```suggestion
    path = "./tests/resources/example_python.py"
```

This needs to be changed back, as the file it is referring to was switched into the resources folder.

sjsikora commented on tests/test_analyzer.py:
```diff
@@ -194,11 +196,11 @@ def test_NaturalLanguageAnalyzer_core_stats(tmp_path: Path):
     """
     This test uses the example_python.py file
-    in the tests/resources/ directory. To ensure
+    in the tests/static/ directory. To ensure
```
```suggestion
    in the tests/resources/ directory. To ensure
```

priyansh1913 commented:
Thanks Sam for catching the issue. I have updated test path to use `tests/resources/` directory and added missing `charset-normalizer` dependency to requirements.txt to ensure the pytest ran. All tests now pass.

sjsikora reviewed (approved):
Thank you, Priyansh.
